src/annotator/integrations/image-text-layer.js

```diff
@@ -194,6 +194,20 @@ export class ImageTextLayer {
     this._listeners.add(window, 'resize', this._updateBoxSizes);
   }
 
+  /**
+   * Synchronously update the text layer to match the size and position of
+   * the image.
```

Comment:
Can you help me understand why a synchronous update is required here? Is it to avoid a scroll-bounce? I do appreciate the clear function naming.

Reply:
In the VS integration, there is code like this:

```js
preserveScrollPosition(() => {
  // Apply styling changes that alter PDF image size

  textLayer.updateSync();
});
```

The `updateSync()` call is needed so that the size of the text layer is updated to reflect the resized image as soon as the callback passed to `preserveScrollPosition` returns. This is important because `preserveScrollPosition` works by measuring the position of some content (the "scroll anchor") before and after it runs the callback, and then adjusting the scroll position so that the scroll anchor remains in the same place within the viewport.